ArcGISRuntimeSDKQt_CppSamples/Analysis/DisplayUtilityAssociations/DisplayUtilityAssociations.cpp

```diff
@@ -0,0 +1,280 @@
+    UniqueValueRenderer* uniqueValueRenderer = new UniqueValueRenderer("", nullptr, QStringList{"AssociationType"}, QList<UniqueValue*>{attachmentValue, connectivityValue}, this);
+    m_associationsOverlay->setRenderer(uniqueValueRenderer);
+
+    // populate the legend
+    m_attachmentSymbol->createSwatch();
+    m_connectivitySymbol->createSwatch();
+  });
+
+  connect(m_utilityNetwork, &UtilityNetwork::associationsCompleted, this, [this](QUuid, const QList<UtilityAssociation*>& associations)
+  {
+    bool matchedGraphic = false;
+    for (UtilityAssociation* association : associations)
+    {
+      // check if the graphics overlay already contains the association
+      for (Graphic* graphic : *m_associationsOverlay->graphics())
+      {
+        if (graphic->attributes()->containsAttribute("GlobalId") && qvariant_cast<QUuid>(graphic->attributes()->operator[]("GlobalId")) == association->globalId())
+        {
+          matchedGraphic = true;
+          break;
+        }
+      }
+
+      // if the association is already in the overlay, don't add a new graphic
+      if (matchedGraphic)
+      {
+        matchedGraphic = false;
+        continue;
+      }
+
+      // add a graphic for the association
+      QVariantMap graphicAttributes;
+      graphicAttributes["GlobalId"] = association->globalId();
+      graphicAttributes["AssociationType"] = static_cast<int>(association->associationType());
+      Graphic* graphic = new Graphic(association->geometry(), graphicAttributes, this);
+
+      m_associationsOverlay->graphics()->append(graphic);
+    }
+  });
+
+  connect(m_attachmentSymbol, &Symbol::createSwatchCompleted, this, [this](QUuid id, QImage image)
+  {
+    if (!m_symbolImageProvider)
+      return;
+
+    // convert the QUuid into a QString
+    const QString imageId = id.toString().remove("{").remove("}");
+
+    // add the image to the provider
+    m_symbolImageProvider->addImage(imageId, image);
+
+    // update the URL with the unique id
+    m_attachmentSymbolUrl = QString("image://%1/%2").arg(SymbolImageProvider::imageProviderId(), imageId);
+
+    // emit the signal to trigger the QML Image to update
+    emit attachmentSymbolUrlChanged();
+  });
+
+  connect(m_connectivitySymbol, &Symbol::createSwatchCompleted, this, [this](QUuid id, QImage image)
+  {
+    if (!m_symbolImageProvider)
+      return;
+
+    // convert the QUuid into a QString
+    const QString imageId = id.toString().remove("{").remove("}");
+
+    // add the image to the provider
+    m_symbolImageProvider->addImage(imageId, image);
+
+    // update the URL with the unique id
+    m_connectivitySymbolUrl = QString("image://%1/%2").arg(SymbolImageProvider::imageProviderId(), imageId);
+
+    // emit the signal to trigger the QML Image to update
+    emit connectivitySymbolUrlChanged();
+
+    m_swatchesCompleted = true;
+    emit swatchesCompletedChanged();
```

Comment:
this seems like a potential timing problem. You're using the 1 signal to notify when both swatches complete. There's no guarantee this one will finish 2nd so the complete signal might actually be early. Maybe the 2 url properties not being empty is enough to detect that the swatches are both complete?